app/src/main/java/com/alexchan/wallpaper/ui/search/searchActivity/SearchActivity.kt

```diff
@@ -0,0 +1,87 @@
+package com.alexchan.wallpaper.ui.search.searchActivity
+
+import android.app.SearchManager
+import android.content.ComponentName
+import android.content.Context
+import android.os.Bundle
+import android.provider.SearchRecentSuggestions
+import android.view.Menu
+import android.view.MenuItem
+import android.view.inputmethod.InputMethodManager
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.SearchView
+import com.alexchan.wallpaper.R
+import com.alexchan.wallpaper.database.SearchSuggestionProvider
+import com.alexchan.wallpaper.ui.search.searchResults.SearchResultsActivity
+import kotlinx.android.synthetic.main.activity_search.*
+
+class SearchActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_search)
+
+        // Set action bar background color -> Replace with Dark Theme support
+        //supportActionBar?.setBackgroundDrawable(ColorDrawable(ContextCompat.getColor(applicationContext, R.color.colorPrimary)))
+
+        // Allow the user to clear recent search history
+        clearSearchHistoryButton.setOnClickListener {clearSearchHistoryConfirmation()}
+    }
+
+    private fun clearSearchHistoryConfirmation() {
+        // Close software keyboard pop up
+        val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
+        imm?.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0)
```

Comment:
Can make this as an extension, in case it will be used on other places. Ignore if this is the only place use this